src/tests/test_blk_read_tracker.cpp

```diff
@@ -352,6 +358,198 @@ TEST_F(BlkReadTrackerTest, TestInsRmWithWaiterOverlapMultiReads2) {
     get_inst()->remove(c);
 }
 
+//////////////////////////// Multi-thread test cases //////////////////////////////
+
+/*
+ * Multi-thread Insert and remove, with no free operation;
+ *
+ * 1. do insert with a few threads -- (can also be done in massive threads, but not necessary)
+ * 2. do remove in massive threads (must be same amount of inserts);
+ * */
+TEST_F(BlkReadTrackerTest, TestThreadedInsertAndRemove) {
+    auto align = 8ul;
+    LOGINFO("Step 1: set entries per record to {}.", align);
+    get_inst()->set_entries_per_record(align);
+
+    std::vector< BlkId > bids{{10, 8, 0}, {20, 5, 0}, {25, 6, 0}, {43, 16, 0}, {56, 4, 0}, {72, 18, 0}, {122, 4, 0}};
+
+    const auto repeat = 100ul;
+    std::vector< std::thread > op_threads;
+
+    for (const auto& b : bids) {
+        std::thread t([this, &b]() {
+            for (auto j = 0ul; j < repeat; ++j) {
+                get_inst()->insert(b);
+            }
+        });
+        op_threads.push_back(std::move(t));
+    }
+
+    LOGINFO("Step 2: threaded insert issued.");
+
+    for (auto& t : op_threads) {
+        t.join();
+    }
+
+    // remove has to wait for insert to complete because if insert thread runs slower than remove, it might assert
+    // complaining no elements are found in map;
```

Comment:
Couldnt we solve this with insert at line 382 also insert to a set and remove thread uses that as a candidate.

Reply:
it is done at the last test case. This test case is not intended for that.